packages/esm-patient-registration-app/src/patient-registration/field/person-attributes/text-person-attribute-field.test.tsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import { render, screen, fireEvent, act } from '@testing-library/react';
+import { TextPersonAttributeField } from './text-person-attribute-field.component';
+import { Form, Formik } from 'formik';
+
+// jest.mock('formik', () => ({
+//   ...jest.requireActual('formik'),
+// }));
+
+jest.mock('formik', () => {
+  const ActualFormik = jest.requireActual('formik');
+  return {
+    ...ActualFormik,
+    Field: ({ name, validate, children }) => {
+      const MockedField = () =>
+        children({
+          field: { name },
+          form: { touched: {}, errors: {} },
+          meta: { error: validate && validate('test-value') },
+        });
+      return <MockedField />;
+    },
+  };
+});
+
+describe('TextPersonAttributeField', () => {
+  const mockPersonAttributeType = {
+    format: 'java.lang.String',
+    display: 'Referred by',
+    uuid: '4dd56a75-14ab-4148-8700-1f4f704dc5b0',
+  };
+
+  it('renders the input field with a label', () => {
+    render(
+      <Formik initialValues={{}} onSubmit={() => {}}>
+        <Form>
+          <TextPersonAttributeField
+            id="attributeId"
+            personAttributeType={mockPersonAttributeType}
+            label="Custom Label"
+          />
+        </Form>
+      </Formik>,
+    );
+
+    expect(screen.getByLabelText(/Custom Label/i)).toBeInTheDocument();
+    expect(screen.getByRole('textbox')).toBeInTheDocument();
+  });
+
+  it('renders the input field with the default label if label prop is not provided', () => {
+    render(
+      <Formik initialValues={{}} onSubmit={() => {}}>
+        <Form>
+          <TextPersonAttributeField id="attributeId" personAttributeType={mockPersonAttributeType} />
+        </Form>
+      </Formik>,
+    );
+
+    expect(screen.getByLabelText(/Referred by/i)).toBeInTheDocument();
+    expect(screen.getByRole('textbox')).toBeInTheDocument();
+  });
+
+  it.skip('validates the input with the provided validationRegex', () => {
```

Comment:
This test is failing because we are not getting the error in the test UI. 

> <body>
      <div>
        <form
          action="#"
        >
          <div
            class="customField halfWidthInDesktopView"
          >
            <div
              style="margin-bottom: 1rem;"
            >
              <div
                class="cds--layer-two"
              >
                <div
                  class="cds--form-item cds--text-input-wrapper"
                >
                  <div
                    class="cds--text-input__label-wrapper"
                  >
                    <label
                      class="cds--label"
                      for="attributeId"
                    >
                      Referred by (optional)
                    </label>
                  </div>
                  <div
                    class="cds--text-input__field-outer-wrapper"
                  >
                    <div
                      class="cds--text-input__field-wrapper"
                    >
                      <input
                        class="cds--text-input cds--text-input--md"
                        id="attributeId"
                        name="attributes.4dd56a75-14ab-4148-8700-1f4f704dc5b0"
                        type="text"
                        value="abc"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </body>

      89 |     });
      90 |     screen.debug();
    > 91 |     expect(screen.getByText(/Invalid Input/i)).toBeInTheDocument();
         |                   ^
      92 |   });
      93 |

This is the ui that is getting rendered. As you can see there is only input field and no errors indicating that input is not valid.